core/src/main/java/brooklyn/catalog/internal/BasicBrooklynCatalog.java

```diff
@@ -750,7 +780,36 @@ private String setFromItemIfUnset(String oldValue, Map<?,?> item, String fieldAt
         return oldValue;
     }
 
-    
+    /** scans the given libraries for annotated items, or if null scans the local classpath */ 
```

Comment:
for non-null is does *not* scan the local classpath.  i've refactored to make 3 methods -- easier to read for sure, but at a cost of more private methods in the class.